quartz/plugins/transformers/ofm.ts

```diff
@@ -222,7 +222,9 @@ export const ObsidianFlavoredMarkdown: QuartzTransformerPlugin<Partial<Options>
                   const ext: string = path.extname(fp).toLowerCase()
                   const url = slugifyFilePath(fp as FilePath)
                   if ([".png", ".jpg", ".jpeg", ".gif", ".bmp", ".svg", ".webp"].includes(ext)) {
-                    const dims = alias ?? ""
+                    const dims = alias?.match(/^\d+(x\d+)?|\|\d+(x\d+)/) ? alias.match(/^\d+(x\d+)?|\|\d+(x\d+)/)![0] : ""
+                    const alt = alias?.replace(dims, "")?.trim()
```

Comment:
This would allow you to capture all the possible things in one match and return easier to identify groups.

```
const imgregex = new RegExp(
    // Test future changes https://rubular.com/r/BnOlfw6C4O4m2j
    // Look for alt
    '(?<alt>' + // Start a named group for alt
        '(?!^\d+$)' + // This look ahead makes sure this isn't just digits
        '.+' + // This searches for any character greedily.
    ')?' + // Close the group. The '?' means it may not be there.

    // Search for digits
    '(^|\|)' + // Either beginning of a line or a pipe
    '\s?+' + // Handle any numbers of spaces after a pipe
    '(?<width>\d+)' + // Start a name group for width
    '(x(?<height>\d+))?' + // Height would require a 'x'. The '?' m
    '$' // Look until the end of the line
)
const matches= alias?.match(imgregex)
const width = matches.width ||= "auto"
const height= matches.height ||= "auto"
const alt= matches.alt ||= "auto"
```

Reply:
Failed to process `content/Compendium/Ambre Loiseau.md`: Invalid regular expression: /^(?<alt>(?!^d+$).*?)?(|?s?+(?<width>d+)(x(?<height>d+))?)?$/: Nothing to repeat
For a file where there is no image found

Reply:
@HeyItsGilbert ruby has a different regex engine from js, you might need to modify it (and test on https://regex101.com/)

Reply:
Regex is valid but looks like capture groups aren't supported. https://www.regular-expressions.info/javascript.html

I'll give a different version when I have some time to update it.

Reply:
Capture group are supported: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_capturing_group

Reply:
Seems this: https://regex101.com/r/lKh0xu/1 should works

Reply:
It's named groups that aren't supported if that article is correct. Without names you'll just grab them by index number. It means we probably should mark some of the groups as non capture groups.

Reply:
This might be because I escaped characters to support `/.../` syntax. That may not be needed for `new RegEsp()`.

Reply:
I think I found it.

```
'\s?+' + // Handle any number of spaces after
```

Should be.
```
'\s+?' + // Handle any number of spaces after
```

The `+?` combo makes it `0 or more` and the `?` makes it optional. I think Javascript didn't like the alternative while other languages accept it. I updated the code above.